client/my-sites/sidebar/utils.js

```diff
@@ -5,18 +5,22 @@ import { includes, some } from 'lodash';
 
 // Check if `path` starts with `pathPrefix` and the very next character
 // (if there is any) is `/` or `?`
-function pathStartsWith( fullPath, pathPrefix ) {
+// If shallow is true, check only the very first fragment.
+function pathStartsWith( fullPath, pathPrefix, shallow ) {
+	if ( shallow && ! includes( [ 'types', 'marketing' ], fullPath.split( '/' )?.[ 1 ] ) ) {
+		return fullPath.split( '/' )?.[ 1 ] === pathPrefix.split( '/' )?.[ 1 ];
+	}
 	return (
 		fullPath === pathPrefix ||
 		( fullPath.slice( 0, pathPrefix.length ) === pathPrefix &&
 			includes( '/?', fullPath[ pathPrefix.length ] ) )
 	);
 }
 
-export function itemLinkMatches( paths, currentPath ) {
+export function itemLinkMatches( paths, currentPath, shallow = false ) {
```

Comment:
Would be great to update the unit tests to support this.

Reply:
Tests incoming!